gdpr/models.py

```diff
@@ -30,18 +30,32 @@ def create_from_purpose_slug(self, purpose_slug, source_object, issued_at=None,
             raise KeyError('Purpose with slug {} does not exits'.format(purpose_slug))
 
     def exists_valid_consent(self, purpose_slug, source_object):
-        return LegalReason.objects.filter(
-            source_object_content_type=ContentType.objects.get_for_model(source_object.__class__),
-            source_object_id=str(source_object.pk),
-            purpose_slug=purpose_slug,
-            is_active=True,
-            expires_at__gte=timezone.now(),
-        ).exists()
+        return LegalReason.objects.filter_source_instance_active_non_expired(
+            source_object).filter(purpose_slug=purpose_slug).exists()
+
+
+class LegalReasonQuerySet(models.QuerySet):
+
+    def filter_active(self):
+        return self.filter(is_active=True)
+
+    def filter_non_expired(self):
```

Comment:
Myslíš že tohle někdo bude používat samostatně?

Reply:
No, tak vzniklo to v době, kdy ještě nebylo jasné, co znamená odebrání consentu, tak jsem je tam pro jistotu dal všechny...